### PR TITLE
[State API] Improve task api

### PR DESCRIPTION
**Pull request by rkooo567**
## Why are these changes needed?

- Add worker id & pg id to the task state
- Add pg id to the actor state
- Add start / end time for worker state
- Add start / end time for node state

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: src/ray/gcs/gcs_server/gcs_worker_manager.cc [hunk 1]
```diff
@@ -25,87 +25,106 @@ void GcsWorkerManager::HandleReportWorkerFailure(
...
-                        "unexpected failures in the cluster.";
-  }
-  auto worker_failure_data = std::make_shared<WorkerTableData>();
-  worker_failure_data->CopyFrom(request.worker_failure());
-  worker_failure_data->set_is_alive(false);
-
-  for (auto &listener : worker_dead_listeners_) {
-    listener(worker_failure_data);
-  }
+  GetWorkerInfo(
```
Comment (rickyyx) [inline src/ray/gcs/gcs_server/gcs_worker_manager.cc:28]: Are these changes mainly on the refactoring of the GetWorkerInfo funcion?

File: python/ray/experimental/state/common.py [hunk 2]
```diff
@@ -416,6 +418,12 @@ class NodeState(StateSchema):
     node_name: str = state_column(filterable=True)
     #: The total resources of the node.
     resources_total: dict = state_column(filterable=False)
+    #: The time when the node (raylet) starts.
```
Comment (rickyyx) [inline python/ray/experimental/state/common.py:421]: When we print out to users on CLI, do you think if we should convert all the timestamps to human readable values?

**Review by rickyyx — COMMENTED** [review]
Looks great to me! Only question is on the gcs worker manager refactoring.

File: python/ray/experimental/state/common.py (hunk 2 above)
Comment (rkooo567) [reply python/ray/experimental/state/common.py:421]: we should. I think it is same for task events. 

I think we need a way to distinguish API return value / display values. Maybe you can consider adding a task to fix this?

File: src/ray/gcs/gcs_server/gcs_worker_manager.cc (hunk 1 above)
Comment (rkooo567) [reply src/ray/gcs/gcs_server/gcs_worker_manager.cc:28]: yes. But also there's a new change that we get WorkerData before we update the failureData. Previously, we overwrote the whole data. Now we merge them.

**Comment by rkooo567** [comment]
cc @iycheng for the worker manager

File: python/ray/experimental/state/common.py (hunk 2 above)
Comment (rickyyx) [reply python/ray/experimental/state/common.py:421]: Right, I guess we could do the formatting upon printing? If I use `list_tasks` programmatically, I might still have the value as `ms`, but I think we should make it readable for display. 
Tracked here: https://github.com/ray-project/ray/issues/31876

Comment (rkooo567) [reply python/ray/experimental/state/common.py:421]: yeah. I think we can either process that from the CLI, or we can have something like view_callback for each field (which is invoked when it is printed)

**Comment by rkooo567** [comment]
also @scv119 for proto changes

**Review by rickyyx — APPROVED** [review]
LGTM!

**Review by scv119 — APPROVED** [review]

**Pull request merged**